[17.0][OU-ADD] analytic: migration to 17.0

hbrunn opened the pull request:
https://github.com/OCA/OpenUpgrade/pull/4458 with review followup

daiduongnguyen-odoo committed:
[OU-ADD] analytic: migration to 17.0

hbrunn committed:
[OU-IMP] analytic: use ir.property api to set properties

and set them on all companies
also save account.analytic.plan#company_id for posterity
and apply noupdate changes

hbrunn commented:
/ocabot migration analytic

<dependency></dependency>

hbrunn commented:
reminder: This needs to load the noupdate changes to be introduced in #4606

pedrobaeza commented on openupgrade_scripts/scripts/analytic/17.0.1.1/upgrade_analysis_work.txt:
```diff
@@ -0,0 +1,26 @@
+analytic     / account.analytic.account / message_main_attachment_id (many2one): DEL relation: ir.attachment
+analytic     / account.analytic.account / root_plan_id (many2one)       : not a function anymore
+analytic     / account.analytic.account / root_plan_id (many2one)       : now related
+analytic     / account.analytic.plan    / _order                        : _order is now 'sequence asc, id' ('complete_name asc')
```
I'm checking now about this point, as we should preserve the order shown in previous version, but I'm not finding right now where the widget is taking the order. Investigating now...

pedrobaeza committed:
[OU-FIX] analytic: Finish migration scripts

- Set account.analytic.plan sequence according previous order (by id)
- Don't put default value for account.analytic.plan.applicability~company_id
  for preserving previous behavior.
- Better work file explanations.

pedrobaeza commented:
Finish migration scripts with these changes:

- Set account.analytic.plan sequence according previous order (by id)
- Don't put default value for account.analytic.plan.applicability~company_id for preserving previous behavior.
- Better work file explanations.
